File: C/cash.c
// Prints minimum number of coins needed to give a change.
// by Wojciech Szubstarski 2018 for CS50.
#include <cs50.h>
#include<math.h>
#include <stdio.h>

float cash;
short penny = 1, nickel = 5, dime = 10, quarter = 25;
int coins = 0;

int main(void)
{
    //Prompts user to enter amount in dollars and validate it.
    do
    {
        cash = get_float("How many $ I owe you? ");
    }
    while (cash < 0);
    //Change dollars to pennies.
    cash *= 100;
    cash = round(cash);
    //Calculating the result.
    while (cash >= quarter)
    {
        cash = cash - quarter;
        coins++;
    }
    while (cash >= dime)
    {
        cash = cash - dime;
        coins++;
    }
    while (cash >= nickel)
    {
        cash = cash - nickel;
        coins++;
    }
    while (cash >= penny)
    {
        cash = cash - penny;
        coins++;
    }
    //Shows number of coins.
    printf("%i\n", coins);
}

File: Python/cash.py
# Prints minimum number of coins needed to give a change.
# by Wojciech Szubstarski 2018 for CS50.

from cs50 import get_float

while True:
    cash = get_float("Change owed: ")
    if cash >= 0:
        break

cash *= 100
penny = 1
nickel = 5
dime = 10
quarter = 25
coins = 0

# Calculating the result.
while (cash >= quarter):
    cash = cash - quarter
    coins += 1

while (cash >= dime):
    cash = cash - dime
    coins += 1

while (cash >= nickel):
    cash = cash - nickel
    coins += 1

while (cash >= penny):
    cash = cash - penny
    coins += 1

# Shows number of coins.
print(coins)
